enumerables.rb

```diff
@@ -0,0 +1,107 @@
+# Enumerable methods
+
+module Enumerable
+  # my_each
+
+  def my_each
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  # end of my_each
+
+  # my_each_with_index
+  def my_each_with_index
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  # end of my_each_with_index
+
+  # my_select
+
+  def my_select
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  # end of my_select
+
+  # my_all?
+
+  def my_all?
+    if_all = true
+    my_each do |i|
+      break if_all = false unless yield(i)
+    end
+    if_all
+  end
+
+  # end of my_all?
+
+  # my_any?
+
+  def my_any?
+    if_any = false
+    my_each do |i|
+      break if_any = true if yield(i)
+    end
+    if_any
+  end
+
+  # end of my_any?
+
+  # my_none?
+
+  def my_none?
+    if_none = true
+    my_each do |i|
+      break if_none = false if yield(i)
+    end
+    if_none
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false, true, []].my_none? == [nil, false, true, []].none? => false
```
- Yours is retrieving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/81133073-c757fa80-8f0d-11ea-9d8e-ea70d5f48173.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690365-cd4ed900-6f68-11ea-9933-f73b0091208d.png)

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81133128-f53d3f00-8f0d-11ea-970f-a0a09dfd2067.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690539-130ba180-6f69-11ea-8bd0-6ca061ab05d2.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81133161-0ede8680-8f0e-11ea-8f2d-43f7e7d3985d.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81133202-261d7400-8f0e-11ea-8474-c82abbcfebb7.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:
```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?